src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -279,11 +279,15 @@ public boolean matches(byte[] buf, int off, int len)
             return false;
         }
 
+        if (temporalId == -1)
+        {
+            return true;
+        }
+
         int tid = track.getMediaStreamTrackReceiver()
             .getStream().getTemporalID(buf, off, len);
 
-        return (tid == -1 && ArrayUtils.isNullOrEmpty(dependencyEncodings))
-                || tid == temporalId;
+        return tid == temporalId;
```

Comment:
Wouldn't that break the case where browser enables simulcast but doesn't use temporal layers (by failing to match packets against any RTPEncoding?). Do we know for example whether firefox enables temporal layers with its simulcast implementation?